edx_repo_tools/oep2/report.py

```diff
@@ -0,0 +1,293 @@
+"""
+``oep2 report``: Check repositories for OEP-2 compliance.
+"""
+
+import logging
+import os.path
+import pkg_resources
+
+import click
+from git.repo.base import Repo
+from git.refs.remote import RemoteReference
+from git.cmd import Git
+import pytest
+import yaml
+from edx_repo_tools.auth import login_github
+
+LOGGER = logging.getLogger(__name__)
+
+
+@click.command()
+@click.option(
+    '-o', '--org',
+    multiple=True,
+    show_default=True,
+    default=[],
+    help="Specify an org to check all of that orgs non-fork "
+         "repositories for OEP-2 compliance",
+)
+@click.option(
+    '-r', '--repo',
+    multiple=True,
+    default=None,
+    help="Specify a repository to check it for OEP-2 compliance",
+)
+@click.option(
+    '--oep',
+    multiple=True, default=None,
+    help="List of OEPs to check for explicit specification of compliance",
+)
+@click.option(
+    '-n', '--num-processes',
+    default=1,
+    help="How many procesess to use while checking repositories",
+)
+@click.option(
+    '--trace/--no-trace', default=False,
+    help="Trace git and github interactions during reporting",
+)
+@click.option(
```

Comment:
Because I need the github connection inside py.test, and I couldn't figure out how to pass it into the `pytest.main` command in a way that it would be accessible. I guess I could smuggle it through a global...